### PR TITLE
Refactor integration tests with pytest

### PR DESCRIPTION
**Pull request by ThomasCluzel**
## Proposed changes

> _Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue._

This pull request refactors the tests. Instead of using bash scripts, we use Python 3 with `pytest`.

## Types of changes

What types of changes does your code introduce to this project?
> _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Issues closed by changes

> _List here all issues closed by your changes. Use a list of items like `- [x] Close #0`_

## Checklist

> _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I agree with the [CODE OF CONDUCT](CODE_OF_CONDUCT.md)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_cnes_sonarqube.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            docker_client.containers.get(cls.SONARQUBE_CONTAINER_NAME).stop()
+
+    def test_up(self):
+        """
+        As a user, I want the server to be UP so that I can use it.
+        """
+        status = requests.get(f"{self.SONARQUBE_URL}/api/system/status",
+                    auth=("admin", self.SONARQUBE_ADMIN_PASSWORD)).json()['status']
+        # Hint: if this test fails, the server might still be starting
+        assert status == "UP"
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:97]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 18
           

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

File: tests/test_cnes_sonarqube.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        for password in ("", "admin"):
+            params = {"name": "tmp", "detach": True}
+            if password:
+                params['environment'] = {"SONARQUBE_ADMIN_PASSWORD": password}
+            docker_client.containers.run("lequal/sonarqube:latest", **params)
+            time.sleep(3)
+            output = docker_client.containers.get("tmp").logs()
+            docker_client.containers.get("tmp").remove(force=True)
+            # Hint: if this test fails, the server may have started with the default admin password or the test did not wait long enough for the expected message to be logged
+            assert b"Failed to start CNES SonarQube." in output
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:251]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
+"""
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:1]: Codacy found an issue: [Multi-line docstring summary should start at the first line (D212)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+            ("SonarTS", "1.9 (build 3766)"),
+            ("SonarVB", "7.15 (build 8572)"),
+            ("SonarXML", "2.0.1 (build 2020)"),
+            ("Svn", "1.9.0.1295")
+        )
+        sonar_plugins = requests.get(f"{self.SONARQUBE_URL}/api/plugins/installed").json()['plugins']
+        installed_plugins = { plugin['name']: plugin['version'] for plugin in sonar_plugins }
+        for name, version in required_plugins:
+            # Hint: if this test fails, one or more plugins may be missing or installed with an outdated version
+            assert installed_plugins[name] == version
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:143]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+Run the tests by launching ``pytest`` from the "tests/" folder.
+
+Pytest documentation: https://docs.pytest.org/en/stable/contents.html
+"""
+
+import os
+import platform
+import re
+import subprocess
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:12]: Codacy found an issue: [Consider possible security implications associated with subprocess module.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+            "CNES_C_EMBEDDED_A",
+            "CNES_C_EMBEDDED_B",
+            "CNES_C_EMBEDDED_C",
+            "CNES_C_EMBEDDED_D"
+        )
+        quality_profiles = requests.get(f"{self.SONARQUBE_URL}/api/qualityprofiles/search").json()['profiles']
+        cnes_quality_profiles = [ qp['name'] for qp in quality_profiles if re.match(r'CNES_\w+_[ABCD]', qp['name']) ]
+        for profile in required_quality_profiles:
+            # Hint: if this test fails, one or more Quality Profiles may be missing, check the container logs
+            assert profile in cnes_quality_profiles
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:187]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        # Check SonarQube logs
+        config_logs = docker_client.containers.get(lequalsonarqube_container_name).logs()
+        subprocess.run("docker-compose down", shell=True, check=True, capture_output=True)
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_data").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_extensions").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_logs").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql_data").remove()
+        # Hint: if this test fails, the server may have been reconfigured when not needed or the test did not wait long enough for the expected message to be logged
+        assert b"[INFO] CNES SonarQube: The database has already been filled with CNES configuration. Not adding anything." in config_logs
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:234]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        if platform.system() == 'Linux':
+            proc_stdout = subprocess.run("sysctl -a", shell=True, check=True, capture_output=True).stdout
+            max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+            # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+            assert max_map_count >= 262144
+        docker_client = docker.from_env()
+        lequalsonarqube_container_name="lequalsonarqube-compose"
+        # Use the compose file with an external database
+        print("Starting the service (sonarqube and postgres)...")
+        subprocess.run("docker-compose up -d", shell=True, check=True, capture_output=True)
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:217]: Codacy found an issue: [subprocess call with shell=True seems safe, but may be changed in the future, consider rewriting without shell](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        subprocess.run("docker-compose down", shell=True, check=True, capture_output=True)
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_data").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_extensions").remove()
+        docker_client.volumes.get("tests_test_volume_compose_sonarqube_logs").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql").remove()
+        docker_client.volumes.get("tests_test_volume_compose_postgresql_data").remove()
+        # Hint: if this test fails, the server may have been reconfigured when not needed or the test did not wait long enough for the expected message to be logged
+        assert b"[INFO] CNES SonarQube: The database has already been filled with CNES configuration. Not adding anything." in config_logs
+
+    def test_no_password_no_run(self):
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:236]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+        As a SonarQube user, I want the configuration of the server
+        not to be executed if the server has already been configured
+        so that the database is not populated more than once.
+        """
+        # On Linux, max_map_count must be at least 262144 to run this test
+        if platform.system() == 'Linux':
+            proc_stdout = subprocess.run("sysctl -a", shell=True, check=True, capture_output=True).stdout
+            max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+            # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+            assert max_map_count >= 262144
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:212]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+        # Wait for the SonarQube container to be configured
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name)
+        # Restart the SonarQube server but not the database
+        print("Restarting SonarQube server...")
+        docker_client.containers.get(lequalsonarqube_container_name).restart()
+        time.sleep(30)
+        self.wait_cnes_sonarqube_ready(lequalsonarqube_container_name, tail=10)
+        # Check SonarQube logs
+        config_logs = docker_client.containers.get(lequalsonarqube_container_name).logs()
+        subprocess.run("docker-compose down", shell=True, check=True, capture_output=True)
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:227]: Codacy found an issue: [subprocess call with shell=True seems safe, but may be changed in the future, consider rewriting without shell](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+
+    def test_check_qg(self):
+        """
+        As a SonarQube user, I want the SonarQube server to have the CNES
+        Quality Gate configured and set as default so that I can use it.
+        """
+        quality_gates = requests.get(f"{self.SONARQUBE_URL}/api/qualitygates/list").json()['qualitygates']
+        cnes_quality_gates = [ gate for gate in quality_gates if gate['name'] == "CNES" ]
+        # Hint: if one of these tests fails, the CNES Quality Gate may not have been added correctly, check the container logs
+        assert cnes_quality_gates # not empty
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:153]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+
+    def test_no_config_twice(self):
+        """
+        As a SonarQube user, I want the configuration of the server
+        not to be executed if the server has already been configured
+        so that the database is not populated more than once.
+        """
+        # On Linux, max_map_count must be at least 262144 to run this test
+        if platform.system() == 'Linux':
+            proc_stdout = subprocess.run("sysctl -a", shell=True, check=True, capture_output=True).stdout
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:209]: Codacy found an issue: [subprocess call with shell=True seems safe, but may be changed in the future, consider rewriting without shell](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+        """
+        Evil User Story:
+        As a hacker, I want to use the default admin password ("admin")
+        to log in as admin.
+        """
+        status = requests.post(f"{self.SONARQUBE_URL}/api/authentication/login",
+                    auth=("admin", "admin"),
+                    data={"login": "admin", "password": "admin"}).status_code
+        # Hint: if this test fails, a hacker may have logged in (HTTP 200) or crash the server (HTTP 500+)
+        assert status == 401 # Unauthorized
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:199]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 13 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:209]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 8 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:217]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 11 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:227]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+    def test_check_qg(self):
+        """
+        As a SonarQube user, I want the SonarQube server to have the CNES
+        Quality Gate configured and set as default so that I can use it.
+        """
+        quality_gates = requests.get(f"{self.SONARQUBE_URL}/api/qualitygates/list").json()['qualitygates']
+        cnes_quality_gates = [ gate for gate in quality_gates if gate['name'] == "CNES" ]
+        # Hint: if one of these tests fails, the CNES Quality Gate may not have been added correctly, check the container logs
+        assert cnes_quality_gates # not empty
+        assert cnes_quality_gates[0]['isDefault']
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:154]: Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 16
           

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

File: tests/test_cnes_sonarqube.py [hunk 16]
```diff
@@ -0,0 +1,238 @@
...
+    if platform.system() == 'Linux':
+        proc_stdout = subprocess.run(["/sbin/sysctl", "-a"], check=True, capture_output=True).stdout
+        max_map_count = int(re.findall('vm.max_map_count = [0-9]+', str(proc_stdout))[0].split(' = ')[1])
+        # Hint: if this test fails, run: sudo sysctl -w vm.max_map_count=262144
+        assert max_map_count >= 262144
+    docker_client = docker.from_env()
+    lequalsonarqube_container_name="lequalsonarqube-compose"
+    # Use the compose file with an external database
+    print("Starting the service (sonarqube and postgres)...")
+    subprocess.run(["docker-compose", "up", "-d"], check=True, capture_output=True)
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:205]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 17]
```diff
@@ -0,0 +1,238 @@
...
+    # Wait for the SonarQube container to be configured
+    TestCNESSonarQube.wait_cnes_sonarqube_ready(lequalsonarqube_container_name)
+    # Restart the SonarQube server but not the database
+    print("Restarting SonarQube server...")
+    docker_client.containers.get(lequalsonarqube_container_name).restart()
+    time.sleep(30)
+    TestCNESSonarQube.wait_cnes_sonarqube_ready(lequalsonarqube_container_name, tail=10)
+    # Check SonarQube logs
+    config_logs = docker_client.containers.get(lequalsonarqube_container_name).logs()
+    subprocess.run(["docker-compose", "down"], check=True, capture_output=True)
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:215]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 16 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:205]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 18]
```diff
@@ -0,0 +1,238 @@
...
+        :param container_name: name of the container running lequal/sonarqube
+        :param tail: forwarded to docker logs
+        """
+        docker_client = docker.from_env()
+        while b'[INFO] CNES SonarQube: ready!' not in docker_client.containers.get(container_name).logs(tail=tail):
+            time.sleep(10)
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the tests
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:57]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 19]
```diff
@@ -0,0 +1,238 @@
...
+    docker_client.volumes.get("tests_test_volume_compose_sonarqube_data").remove()
+    docker_client.volumes.get("tests_test_volume_compose_sonarqube_extensions").remove()
+    docker_client.volumes.get("tests_test_volume_compose_sonarqube_logs").remove()
+    docker_client.volumes.get("tests_test_volume_compose_postgresql").remove()
+    docker_client.volumes.get("tests_test_volume_compose_postgresql_data").remove()
+    # Hint: if this test fails, the server may have been reconfigured when not needed or the test did not wait long enough for the expected message to be logged
+    assert b"[INFO] CNES SonarQube: The database has already been filled with CNES configuration. Not adding anything." in config_logs
+
+def test_no_password_no_run():
+    """As a SonarQube user, I want the container not to start when I forget to
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:225]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 20]
```diff
@@ -0,0 +1,238 @@
...
+    # Class variables
+    RUN = os.environ.get('RUN', "yes")
+    SONARQUBE_CONTAINER_NAME = os.environ.get('SONARQUBE_CONTAINER_NAME', "lequalsonarqube")
+    SONARQUBE_ADMIN_PASSWORD = os.environ.get('SONARQUBE_ADMIN_PASSWORD', "adminpassword")
+    SONARQUBE_URL = os.environ.get('SONARQUBE_URL', "http://localhost:9000")
+
+    # Functions
+    @classmethod
+    def wait_cnes_sonarqube_ready(cls, container_name: str, tail = "all"):
+        """This function waits for SonarQube to be configured by
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:45]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 21]
```diff
@@ -0,0 +1,238 @@
...
+import re
+import subprocess
+import time
+
+import docker
+import requests
+
+
+class TestCNESSonarQube:
+    """This class test the lequal/sonarqube image.
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:19]: Codacy found an issue: [1 blank line required before class docstring (found 0) (D203)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 22]
```diff
@@ -0,0 +1,238 @@
...
+            "CNES_C_EMBEDDED_D"
+        )
+        quality_profiles = requests.get(f"{self.SONARQUBE_URL}/api/qualityprofiles/search").json()['profiles']
+        cnes_quality_profiles = [ qp['name'] for qp in quality_profiles if re.match(r'CNES_\w+_[ABCD]', qp['name']) ]
+        for profile in required_quality_profiles:
+            # Hint: if this test fails, one or more Quality Profiles may be missing, check the container logs
+            assert profile in cnes_quality_profiles
+
+    def test_eus_admin(self):
+        """Evil User Story:
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:179]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 23]
```diff
@@ -0,0 +1,238 @@
...
+
+
+def test_no_config_twice():
+    """As a SonarQube user, I want the configuration of the server
+    not to be executed if the server has already been configured
+    so that the database is not populated more than once.
+    """
+    # On Linux, max_map_count must be at least 262144 to run this test
+    if platform.system() == 'Linux':
+        proc_stdout = subprocess.run(["/sbin/sysctl", "-a"], check=True, capture_output=True).stdout
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:197]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 24]
```diff
@@ -0,0 +1,238 @@
+"""Automated integration test of CNES SonarQube
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:1]: Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 25]
```diff
@@ -0,0 +1,238 @@
...
+        """
+        status = requests.post(f"{self.SONARQUBE_URL}/api/authentication/login",
+                    auth=("admin", "admin"),
+                    data={"login": "admin", "password": "admin"}).status_code
+        # Hint: if this test fails, a hacker may have logged in (HTTP 200) or crash the server (HTTP 500+)
+        assert status == 401 # Unauthorized
+
+
+def test_no_config_twice():
+    """As a SonarQube user, I want the configuration of the server
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:191]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 26]
```diff
@@ -0,0 +1,238 @@
...
+                ports={9000: 9000})
+        else:
+            print(f"Using container {cls.SONARQUBE_CONTAINER_NAME}")
+        # Wait for the SonarQube server inside it to be set up
+        print(f"Waiting for {cls.SONARQUBE_CONTAINER_NAME} to be up...")
+        cls.wait_cnes_sonarqube_ready(cls.SONARQUBE_CONTAINER_NAME)
+
+    @classmethod
+    def teardown_class(cls):
+        """Stop the container"""
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:78]: Codacy found an issue: [First line should end with a period, question mark, or exclamation point (not 'r') (D415)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 27]
```diff
@@ -0,0 +1,238 @@
...
+        Quality Gate configured and set as default so that I can use it.
+        """
+        quality_gates = requests.get(f"{self.SONARQUBE_URL}/api/qualitygates/list").json()['qualitygates']
+        cnes_quality_gates = [ gate for gate in quality_gates if gate['name'] == "CNES" ]
+        # Hint: if one of these tests fails, the CNES Quality Gate may not have been added correctly, check the container logs
+        assert cnes_quality_gates # not empty
+        assert cnes_quality_gates[0]['isDefault']
+
+    def test_check_qp(self):
+        """As a SonarQube user, I want the SonarQube server to have the
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:147]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py [hunk 28]
```diff
@@ -0,0 +1,238 @@
...
+
+    def test_up(self):
+        """As a user, I want the server to be UP so that I can use it."""
+        status = requests.get(f"{self.SONARQUBE_URL}/api/system/status",
+                    auth=("admin", self.SONARQUBE_ADMIN_PASSWORD)).json()['status']
+        # Hint: if this test fails, the server might still be starting
+        assert status == "UP"
+
+    def test_check_plugins(self):
+        """As a SonarQube user, I want the plugins listed in the README
```
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:92]: Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 17 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py:215]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

File: tests/test_cnes_sonarqube.py (hunk 16 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 17 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 16 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 23 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 17 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [Starting a process with a partial executable path](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

File: tests/test_cnes_sonarqube.py (hunk 16 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 23 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 17 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

File: tests/test_cnes_sonarqube.py (hunk 16 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 23 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

File: tests/test_cnes_sonarqube.py (hunk 17 above)
Comment (ThomasCluzel) [inline tests/test_cnes_sonarqube.py]: Codacy found an issue: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

**Comment by ThomasCluzel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- tests/test_cnes_sonarqube.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/cnescatlab/sonarqube/pullRequest?prid=6252252&bid=20358714)

**Review by begarco — APPROVED** [review]

**Review by begarco — APPROVED** [review]

**Pull request merged**